src/http/cookie.cr

```diff
@@ -25,23 +26,27 @@ module HTTP
     def_equals_and_hash name, value, path, expires, domain, secure, http_only
 
     def initialize(@name : String, value : String, @path : String = "/",
-                   @expires : Time? = nil, @domain : String? = nil,
-                   @secure : Bool = false, @http_only : Bool = false,
-                   @samesite : SameSite? = nil, @extension : String? = nil)
+                   @expires : Time? = nil, @max_age : Time::Span? = nil,
+                   @domain : String? = nil, @secure : Bool = false,
+                   @http_only : Bool = false, @samesite : SameSite? = nil,
+                   @extension : String? = nil)
+      @creation_time = Time.now
       @name = URI.unescape name
       @value = URI.unescape value
     end
 
     def to_set_cookie_header
       path = @path
       expires = @expires
+      max_age = @max_age
       domain = @domain
       samesite = @samesite
       String.build do |header|
         header << "#{URI.escape @name}=#{URI.escape value}"
         header << "; domain=#{domain}" if domain
         header << "; path=#{path}" if path
         header << "; expires=#{HTTP.format_time(expires)}" if expires
+        header << "; max-age=#{max_age.total_seconds}" if max_age
```

Comment:
I think `total_seconds` it a `Float64`. Does `max-age` allow float values in the RFC?

Reply:
According to the RFC, Max-Age has to be a non-zero digit. It doesn't say anything about Float vs Int though.

Reply:
https://tools.ietf.org/html/rfc6265#section-5.2.2

Mentions:

```
   Let delta-seconds be the attribute-value converted to an integer.

   If delta-seconds is less than or equal to zero (0), let expiry-time
   be the earliest representable date and time.  Otherwise, let the
   expiry-time be the current date and time plus delta-seconds seconds.

   Append an attribute to the cookie-attribute-list with an attribute-
   name of Max-Age and an attribute-value of expiry-time.
```
so looks like it should be converted to an int.

Reply:
Seems like `max_age.to_i` (number of full seconds) should be enough.

Reply:
At the very least I feel like it should be an `Int64`. I have run into cases where Max-Age is too large and can't be parsed into an `Int32`.

Reply:
@watzon [`Time::Span#to_i`](https://crystal-lang.org/api/0.29.0/Time/Span.html#to_i%3AInt64-instance-method) returns an `Int64`, so no problem here.

Reply:
Oh it does? Well awesome then

Reply:
Done